ingestion_tools/scripts/tomoman_to_tsv.py

```diff
@@ -214,24 +225,71 @@ def cli(ctx):
     pass
 
 
+def mark_exists_in_s3(row, fs, in_s3, not_in_s3):
+    gain_ref = row["frame_gain_reference"]
+    if gain_ref in in_s3:
+        return
+    if gain_ref in not_in_s3:
+        not_in_s3[gain_ref] += 1
+        return
+    # The gain_ref filenames are named with the date in them and are hence distinct. In some cases the file path
+    # might be incorrect but the filename might match a file existing in s3. So, we are also tracking the filename and
+    # the correct path for it.
+    gain_filename_ref = gain_ref.split("/")[-1]
+    if gain_filename_ref in in_s3:
+        return
+    full_path = os.path.join("cryoetportal-rawdatasets-dev/chlamy2/data/chlamy_visual_proteomics", gain_ref)
+    if fs.glob(full_path):
+        in_s3[gain_ref] = gain_ref
```

Comment:
I don't think we need to store this version in our cache at all -- just the mapping from the filename to the *correct* path is fine, and that also means that `get_gain_ref` only needs to check for the filename.